pilot/pkg/networking/util/util.go

```diff
@@ -276,6 +276,11 @@ func IsXDSMarshalingToAnyEnabled(node *model.Proxy) bool {
 	return !features.DisableXDSMarshalingToAny
 }
 
+// IsProtocolSniffingEnabled checks whether protocol sniffing is enabled.
+func IsProtocolSniffingEnabled(node *model.Proxy) bool {
```

Comment:
Please add a node meta as well - global settings pushed to the entire cluster are a critical danger.

Reply:
Done.